### PR TITLE
Homework6 - done

### PR DESCRIPTION
**Pull request by Nikolaenko94**
@vitalics  homework6 -done

### CODE REVIEW

File: hw5/task1/calculator.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+emitter.on('minus', minus.minus);
+//CULCULATION START
+let myInput = fs.createReadStream(__dirname + '/input.txt', 'utf8');
+
+let funcInput = myInput.on('data', (value)=>{
+    let arrInputValue = value.split('\r\n');
+    arrInputValue.forEach(element => {
+        if(element.includes('+')){
+            let arr = element.split(' ');    
+            result.result.emit('result', emitter.emit('plus', arr[0],arr[1]));
```
Comment (vitalics) [inline hw5/task1/calculator.js:26]: too many result.result strings

File: hw6/task2/loading.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+//     })
+//     .then(()=>{
+//         console.clear();
+//         console.log("I replace loading symbols");
+//         return loading(500);
+//     });
+
+//The second way
+asyncFn = function (){
+            setTimeout(() => {
```
Comment (vitalics) [inline hw6/task2/loading.js]: too many nested `setTimeout`. Try to optimize it.
P.S. recursive call may helps you
P.P.S Thing about the long operations. You must see many repeatable loader states.

File: hw6/task1/library.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    next(){
+        this.nextTrack();
+    };
+    prev(){
+        this.prevTrack();
+    };
+    pause(){
+        this.pauseTrack();
+    };
+    search(value){
```
Comment (vitalics) [inline hw6/task1/library.js:183]: search should work in entities that already has in Library class. that means - you can iterate evety of your arrays([see declaration line](https://github.com/Nikolaenko94/Auto-JS-2021/pull/6/files#diff-0ed900fd14b64e674fdf8e7dc523034d9c63dcebe14c33dd2981aeca1c5f266eR4)). and create a new array for finded entities. [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method can helps you

**Review by vitalics — CHANGES_REQUESTED** [review]

File: hw6/task2/loading.js (hunk 2 above)
Comment (Nikolaenko94) [reply hw6/task2/loading.js]: Fixed.

`let i =0 ;` 
`const asyncFn = function(){    `
            `setTimeout(function() {`
           ` let symbols = ["|","/","-","\\"];`   
           ` console.clear();`
           ` console.log(symbols[i]);`
           ` i++;`
            `if(i === symbols.length){return}`
            `asyncFn();   `
`}, 500);`
`};`

File: hw6/task1/library.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    Album = [];
+    Artist = [];
+ // ADD METHOD WITH NESTINGS   
+    add(...args){ 
+    //    value.constuctor.name; - return Class name "Track,Artist,Album";
+    //    value.name - return class.name "Linkin park exemple";
+    // IF :for classes! if one value - this CLASS. if more value: first - Class, nexts - parameters this Class;
+    //ELSE: For class parameters. If we have more then one parameters = > [new Class,...parametrs for this class ];
+        let myArguments = [];
+        myArguments.push(...args);
```
Comment (vitalics) [inline hw6/task1/library.js:14]: `args` is already an array since you're using spread operator

File: hw6/task1/library.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+ // ADD METHOD WITH NESTINGS   
+    add(...args){ 
+    //    value.constuctor.name; - return Class name "Track,Artist,Album";
+    //    value.name - return class.name "Linkin park exemple";
+    // IF :for classes! if one value - this CLASS. if more value: first - Class, nexts - parameters this Class;
+    //ELSE: For class parameters. If we have more then one parameters = > [new Class,...parametrs for this class ];
+        let myArguments = [];
+        myArguments.push(...args);        
+                            // TRACK                            
+            if(myArguments[0].constructor.name === "Track"){
```
Comment (vitalics) [inline hw6/task1/library.js:16]: why not use `instanceof` operator?

File: hw6/task1/library.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }        
+    }
+    
+//PLAY METHOD
+    play(){ 
+        const EventEmitter = require('events');
```
Comment (vitalics) [inline hw6/task1/library.js:125]: all imports should be declared at the start of file, no matter where we use this variable

File: hw6/task2/loading.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            console.clear();
+            console.log(symbols[i]);
+            i++;
+            if(i === symbols.length){return}
+            asyncFn();   
+}, 500);
+};
+
+const loading = (func) => new Promise(
+    (resolve) => setTimeout(() => {
```
Comment (vitalics) [inline hw6/task2/loading.js:50]: resolve is never used. Promise will never get the result

File: hw6/task2/loading.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        console.clear();
+        console.log("I replace loading symbols");
+    }, 2500)
+);
+
+module.exports = {
```
Comment (vitalics) [inline hw6/task2/loading.js:56]: only 1 function should be exportable. I just want to use api such as:
``` js
loading().then(() => console.log('all done'))
```

**Review by vitalics — CHANGES_REQUESTED** [review]

**Pull request merged**